pthread/pthread.c

```diff
@@ -996,26 +1033,23 @@ int pthread_cond_timedwait(pthread_cond_t *__restrict cond,
 	const struct timespec *__restrict abstime)
 {
 	int err = 0;
-	struct timespec now;
-	clock_gettime(cond->clock_id, &now);
-	if ((now.tv_sec > abstime->tv_sec) || ((now.tv_sec == abstime->tv_sec) && (now.tv_nsec >= abstime->tv_nsec))) {
-		err = -ETIMEDOUT;
+
+	const time_t abstime_us = timespec_to_us(abstime);
+
+	if (abstime_us == 0) {
```

Comment:
Perhaps add comment why this is needed - somebody might fell urge to remove this, as "it will timeout anyway"